### PR TITLE
feat: set origin for walletconnect rpc url

### PR DESCRIPTION
**Pull request by rkolpakov**
WalletConnect requires to have a full URL (with origin part) to use the reverse deep link.

### CODE REVIEW

File: packages/web3-react/src/context/connectors.tsx [hunk 1]
```diff
@@ -44,6 +44,14 @@ const ProviderConnectors: FC<ConnectorsContextProps> = (props) => {
   } = props;
 
   const { supportedChainIds } = useSDK();
+  const walletConnectRPC = useMemo(
+    () =>
+      Object.entries(rpc).reduce(
+        (result, [key, value]) => ({ ...result, [key]: BASE_URL + value }),
```
Comment (avsetsin) [inline packages/web3-react/src/context/connectors.tsx:50]: It seems that `value` can be both an absolute and a relative url. So in the case of an absolute url it is worth to add some extra check

**Review by avsetsin — APPROVED** [review]
LGTM. The regexp does not work with cyrillic domains, but I don't think this is a problem in our case.

File: packages/web3-react/src/helpers/regex.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const pattern = new RegExp(
+  /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/,
+);
+
+export const isUrl = (url: string): boolean => pattern.test(url);
```
Comment (alx-khramov) [inline packages/web3-react/src/helpers/regex.ts:5]: It is not obvious why we have `isUrl` checker in the file named `regex`. I assume that this file is intended for adding some another regex-check-functions in the future, but maybe it is better to separate such checkers to separate files.
So, this file can be named as `isUrl.ts` and the export can be default.

File: packages/web3-react/src/helpers/regex.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const pattern = new RegExp(
```
Comment (alx-khramov) [inline packages/web3-react/src/helpers/regex.ts:1]: I guess it is not critical at all, but maybe it is better to use this regex for urls:
https://gist.github.com/dperini/729294
You can see some tests here: https://mathiasbynens.be/demo/url-regex

Comment (mymphe) [reply packages/web3-react/src/helpers/regex.ts:1]: According to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#extracting_sub-domain_name_from_url), it is better to use `URL` constructor to test urls for validity, e.g.
```
function isValidUrl(url) {
  try {
    new URL(url)
    return true
  } catch {
    return false
  }
}
```

**Review by mymphe — APPROVED** [review]
LGTM

File: packages/web3-react/src/helpers/regex.ts (hunk 2 above)
Comment (rkolpakov) [reply packages/web3-react/src/helpers/regex.ts:5]: Totally agree.

**Pull request merged**